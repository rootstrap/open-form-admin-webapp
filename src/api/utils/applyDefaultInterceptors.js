import { saveSession, logout } from 'actions/userActions';

const ACCESS_TOKEN = 'access-token';
const UID = 'uid';
const CLIENT = 'client';
const RS_API_KEY = 'RS-API-KEY';

const UNAUTHORIZED = 401;

const defaultRequestInterceptors = store => [
  async request => {
    try {
      const { info } = store.getState().session;
      if (info) {
        const { token, client, uid } = info;
        request.headers = {
          ...request.headers,
          [ACCESS_TOKEN]: token,
          [RS_API_KEY]: process.env.RS_API_KEY,
          client,
          uid
        };
      }
    } catch (error) {
      console.log('Failed to load session headers', error); // eslint-disable-line
    }
    return request;
  }
];

const defaultResponseInterceptors = store => [
  async response => {
    if (response.ok) {
      const { headers } = response;
      const token = headers.get(ACCESS_TOKEN);
      if (token) {
        const session = {
          token,
          uid: headers.get(UID),
          client: headers.get(CLIENT)
        };
        store.dispatch(saveSession(session));
      }
    }
    if (response.status === UNAUTHORIZED) {
      store.dispatch(logout());
    }
    return response;
  }
];

export default (store, apiService) => {
  defaultRequestInterceptors(store).forEach(interceptor =>
    apiService.requestInterceptors.use(interceptor)
  );

  defaultResponseInterceptors(store).forEach(interceptor =>
    apiService.responseInterceptors.use(interceptor)
  );
};
