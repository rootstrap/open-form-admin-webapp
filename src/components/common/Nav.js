import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

import mobile from 'utils/styles/mobile';
import routes from 'constants/routesPaths';
import Logo from 'components/common/Logo';
import DropdownArrow from 'components/common/DropdownArrow';
import List from 'components/common/List';
import Link from 'components/common/Link';
import { transitionHeight } from 'utils/styles/transition';

const StyledNav = styled.nav`
  display: flex;
  align-items: center;
  background: ${({ theme }) => theme.background.primary};
  height: 3rem;

  ${mobile(css`
    justify-content: space-between;
    position: fixed;
    width: 100%;
    left: 0;
    top: 0;
  `)}
`;

const NavList = styled(List)`
  flex-direction: row;
  align-items: center;

  & > li {
    display: inline-block;
    margin: 0 1rem;
  }

  ${mobile(css`
    position: absolute;
    top: 3.2rem;
    left: 0;
    width: 100%;
    margin: 0;
    overflow: hidden;
    height: 2.5rem;
    ${transitionHeight('2.5rem', '250ms')}

    ${({ theme }) => css`
      background-color: ${theme.background.primary};
      border: 1px solid ${theme.background.primary};
    `}
  `)}
`;

const LogoLink = styled(Link)`
  padding-left: 1rem;
`;

const Menu = styled.button`
  display: none;
  border: none;
  outline: none;
  cursor: pointer;
  background: none;
  position: relative;
  top: 0;
  left: 0;
  height: 1.5rem;
  width: 1.5rem;
  margin-right: 1rem;

  ${mobile(css`
    display: ${({ authenticated }) => (authenticated ? 'block' : 'none')};
  `)}
`;

const DesktopContainer = styled.div`
  display: ${({ authenticated }) => (authenticated ? 'block' : 'none')};

  ${mobile(css`
    display: none;
  `)}
`;

const StyledDropdownArrow = styled(DropdownArrow)`
  height: 1.5rem;
  width: 1.5rem;
  position: absolute;
  top: 0;
  left: 0;
`;

const Nav = ({ children, authenticated }) => {
  const [showMenu, setShowMenu] = useState(false);
  return (
    <StyledNav>
      <LogoLink to={routes.index}>
        <Logo />
      </LogoLink>
      <DesktopContainer authenticated={authenticated}>
        <NavList>{children}</NavList>
      </DesktopContainer>
      <Menu onClick={() => setShowMenu(!showMenu)} authenticated={authenticated}>
        <StyledDropdownArrow active={showMenu} />
      </Menu>
      <CSSTransition in={showMenu} timeout={250} unmountOnExit>
        <NavList>{children}</NavList>
      </CSSTransition>
    </StyledNav>
  );
};

Nav.propTypes = {
  children: PropTypes.node.isRequired,
  authenticated: PropTypes.bool
};

Nav.defaultProps = {
  authenticated: false
};

export default Nav;
