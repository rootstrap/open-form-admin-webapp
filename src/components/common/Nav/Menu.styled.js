import styled, { css } from 'styled-components';
import mobile from 'utils/styles/mobile';

export default styled.button`
  display: none;
  border: none;
  outline: none;
  cursor: pointer;
  background: none;
  position: relative;
  top: 0;
  left: 0;
  height: 1.5rem;
  width: 1.5rem;
  margin-right: 1rem;

  ${mobile(css`
    display: ${({ authenticated }) => (authenticated ? 'block' : 'none')};
  `)}
`;
