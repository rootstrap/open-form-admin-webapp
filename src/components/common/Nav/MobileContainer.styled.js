import styled, { css } from 'styled-components';

import mobile from 'utils/styles/mobile';
import { transitionHeight } from 'utils/styles/transition';

export default styled.div`
  display: none;

  ${mobile(css`
    position: absolute;
    top: 3.2rem;
    left: 0;
    width: 100%;
    margin: 0;
    overflow: hidden;
    height: 2.5rem;
    display: flex;
    ${transitionHeight('2.5rem', '250ms')}

    ${({ theme }) => css`
      background-color: ${theme.background.primary};
      border: 1px solid ${theme.background.primary};
    `}
  `)}
`;
