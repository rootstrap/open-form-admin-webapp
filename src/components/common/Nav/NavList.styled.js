import styled, { css } from 'styled-components';

import mobile from 'utils/styles/mobile';
import List from 'components/common/List';
import { transitionHeight } from 'utils/styles/transition';

export default styled(List)`
  flex-direction: row;
  align-items: center;

  & > li {
    display: inline-block;
    margin: 0 1rem;
  }

  ${mobile(css`
    position: absolute;
    top: 3.2rem;
    left: 0;
    width: 100%;
    margin: 0;
    overflow: hidden;
    height: 2.5rem;
    ${transitionHeight('2.5rem', '250ms')}

    ${({ theme }) => css`
      background-color: ${theme.background.primary};
      border: 1px solid ${theme.background.primary};
    `}
  `)}
`;
