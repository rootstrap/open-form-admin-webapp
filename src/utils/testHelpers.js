import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from 'constants/theme';

export function renderWithTheme(component) {
  return TestRenderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}
