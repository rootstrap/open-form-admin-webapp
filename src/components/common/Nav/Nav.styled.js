import styled, { css } from 'styled-components';

import mobile from 'utils/styles/mobile';
import { DropdownArrow, Link, List } from 'components/common';

export const LogoLink = styled(Link)`
  padding-left: 1rem;
`;

export const DesktopContainer = styled.div`
  display: block;

  ${mobile(css`
    display: none;
  `)}
`;

export const NavList = styled(List)`
  flex-direction: row;
  align-items: center;

  & > li {
    display: inline-block;
    margin: 0 1rem;
  }
`;

export const StyledDropdownArrow = styled(DropdownArrow)`
  height: 1.5rem;
  width: 1.5rem;
  position: absolute;
  top: 0;
  left: 0;
`;

export default styled.nav`
  display: flex;
  align-items: center;
  background: ${({ theme }) => theme.background.primary};
  height: 3rem;

  ${mobile(css`
    justify-content: space-between;
    position: fixed;
    width: 100%;
    left: 0;
    top: 0;
  `)}
`;
