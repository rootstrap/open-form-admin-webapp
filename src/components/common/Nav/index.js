import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

import routes from 'constants/routesPaths';
import { Logo } from 'components/common';
import Menu from './Menu.styled';
import MobileContainer from './MobileContainer.styled';
import StyledNav, { LogoLink, DesktopContainer, StyledDropdownArrow, NavList } from './Nav.styled';

const Nav = ({ children, authenticated }) => {
  const [showMenu, setShowMenu] = useState(false);
  return (
    <StyledNav>
      <LogoLink to={routes.index}>
        <Logo />
      </LogoLink>
      <DesktopContainer authenticated={authenticated}>
        <NavList>{children}</NavList>
      </DesktopContainer>
      <Menu onClick={() => setShowMenu(!showMenu)} authenticated={authenticated}>
        <StyledDropdownArrow active={showMenu} />
      </Menu>
      <CSSTransition in={showMenu} timeout={250} unmountOnExit>
        <MobileContainer>
          <NavList>{children}</NavList>
        </MobileContainer>
      </CSSTransition>
    </StyledNav>
  );
};

Nav.propTypes = {
  children: PropTypes.node.isRequired,
  authenticated: PropTypes.bool
};

Nav.defaultProps = {
  authenticated: false
};

export default Nav;
