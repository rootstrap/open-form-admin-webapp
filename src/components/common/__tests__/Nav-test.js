import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { renderWithTheme } from 'utils/testHelpers';
import Nav from '../Nav';

it('renders correctly', () => {
  const testRenderer = renderWithTheme(
    <Router>
      <Nav>navigation</Nav>
    </Router>
  );
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
