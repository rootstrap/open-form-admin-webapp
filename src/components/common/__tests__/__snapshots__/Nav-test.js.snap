// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c4 {
  display: none;
  border: none;
  outline: none;
  cursor: pointer;
  background: none;
  position: relative;
  top: 0;
  left: 0;
  height: 1.5rem;
  width: 1.5rem;
  margin-right: 1rem;
}

.c5.enter {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5.enter-active {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  -webkit-transition: -webkit-transform 250ms;
  -webkit-transition: transform 250ms;
  transition: transform 250ms;
}

.c5.enter-done {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c5.exit {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c5.exit-active {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 250ms;
  -webkit-transition: transform 250ms;
  transition: transform 250ms;
}

.c1 {
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.1em 0.2em;
  color: #56ccf2;
  padding-left: 1rem;
}

.c2 {
  display: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
  margin: 0 0.75em;
  padding: 0;
  word-wrap: break-word;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > li {
  display: inline-block;
  margin: 0 1rem;
}

.c6 {
  height: 1.5rem;
  width: 1.5rem;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f2f2f2;
  height: 3rem;
}

@media (max-width:50em) {
  .c4 {
    display: none;
  }
}

@media (max-width:50em) {
  .c2 {
    display: none;
  }
}

@media (max-width:50em) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    position: fixed;
    width: 100%;
    left: 0;
    top: 0;
  }
}

<nav
  className="c0"
>
  <a
    className="c1"
    href="/"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="21"
      viewBox="0 0 51 21"
      width="51"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M47.111 0L30.483 15.9948C27.8207 9.20742 20.5304 5.18777 13.0963 6.40841C5.66223 7.62905 0.170853 13.7474 0 21H5.50501C5.69649 15.8218 10.021 11.669 15.4045 11.4937C20.788 11.3183 25.3946 15.1802 25.9498 20.3341L25.8961 20.3964C25.9831 20.6279 25.9998 20.8219 26.0053 21H32.9745L51 3.70002L47.111 0Z"
        fill="#161631"
      />
    </svg>
  </a>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      navigation
    </ul>
  </div>
  <button
    className="c4"
    onClick={[Function]}
  >
    <svg
      className="c5 c6"
      fill="none"
      height="8"
      viewBox="0 0 15 8"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.48242 1L7.48242 7L13.4824 1"
        stroke="#4F4F4F"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</nav>
`;
